src/Internal/Testnet/Contract.purs

```diff
@@ -0,0 +1,375 @@
+                getWalletAddress
+              withCardanoCliUtxos genesisAddr do
+                constraints <- liftAff $ fold <$> traverse
+                  ( \{ wallet, amount } -> do
+                      addrs <- (unwrap wallet).address network
+                      pure $ mustPayToAddress addrs $ Value.lovelaceValueOf
+                        amount
+                  )
+                  utxos
+
+                txHash <- submitTxFromConstraints mempty constraints
+                logInfo' $ "FundWalletsFromGenesis txHash: " <> show txHash
+                awaitTxConfirmed txHash
+        )
+    )
+    roundsFixed
+
+makeDistrFundsPlan
+  :: forall (distr :: Type) (wallets :: Type)
+   . UtxoDistribution distr wallets
+  => (forall a. Address -> Contract a -> Contract a)
+  -> Array KeyWallet
+  -> distr
+  -> Contract (wallets /\ Array (Array (DistrFunds.Tx KeyWallet BigInt)))
+makeDistrFundsPlan withCardanoCliUtxos genesisWallets distr = do
+  let distrArray = map BigNum.toBigInt <$> encodeDistribution distr
+  privateKeys <-
+    for (encodeDistribution distr) \_ ->
+      liftEffect $ wrap <$> Csl.privateKey_generateEd25519
+  wallets <-
+    liftContractM
+      "Impossible happened: could not decode wallets. Please report as bug"
+      $ decodeWallets distr privateKeys
+  let
+    kws = keyWallets (Proxy :: _ distr) wallets
+    targets = Array.concat $ sequence <$> Array.zip kws distrArray
+  sources <- Array.concat <$>
+    parTraverse (\kw -> map (Tuple kw) <$> getGenesisUtxos kw)
+      genesisWallets
+  distrPlan <-
+    liftContractE $
+      DistrFunds.makeDistributionPlan distrFundsParams sources targets
+  pure $ wallets /\ distrPlan
+  where
+  getGenesisUtxos :: KeyWallet -> Contract (Array BigInt)
+  getGenesisUtxos genesisWallet =
+    withKeyWallet genesisWallet do
+      genesisAddr <- liftedM "Could not get genesis address" getWalletAddress
+      withCardanoCliUtxos genesisAddr do
+        liftedM "Could not get genesis wallet utxos" getWalletUtxos
+          <#> map
+            ( BigNum.toBigInt
+                <<< unwrap
+                <<< Value.getCoin
+                <<< _.amount
+                <<< unwrap
+            )
+          <<< Array.fromFoldable
+          <<< Map.values
+
+-- FIXME: adjust values
+distrFundsParams :: DistrFundsParams KeyWallet BigInt
+distrFundsParams =
+  { maxRounds: 3
+  , maxUtxosPerTx: 100
+  , getUtxoMinAdaForWallet: const zero
+  , feePerTx: BigInt.fromInt 2_000_000
+  }
+
+-- FIXME: move to helpers
+mustPayToAddress :: Address -> Value -> TxConstraints
+mustPayToAddress addr =
+  let
+    mSkh = case getStakeCredential addr of
+      Just (StakeCredential (PubKeyHashCredential skh')) -> Just $
+        StakePubKeyHash skh'
+      _ -> Nothing
+  in
+    case getPaymentCredential addr of
+      Just (PaymentCredential (PubKeyHashCredential pkh)) ->
+        case mSkh of
+          Nothing ->
+            Constraints.mustPayToPubKey $ wrap pkh
```

Comment:
These are now deprecated with a type-level warning. Please use the underlying data constructors so that the warning is not propagated to the call site